test/logging/tlogger.go

```diff
@@ -0,0 +1,358 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// TLogger is TLogger
+type TLogger struct {
+	l     *zap.Logger
```

Comment:
`t` is okay here, but I feel changing to `logger`, `errs` and `writer` will be more readable here.

Reply:
https://github.com/golang/go/wiki/CodeReviewComments#variable-names

Reply:
Yes, but there is also
```
The basic rule: the further from its declaration that a name is used, the more descriptive the name must be. 
```

The fields are actually used in multiple other functions that can make readers confusing. e.g. when I read `o.w.Disable()` in the very end of this file, I need to go back to see what is `w`.

Reply:
Compromised!